lib/dns/index.js

```diff
@@ -0,0 +1,250 @@
+ * servers.
+ *
+ * @constructor
+ * @alias module:dns
+ *
+ * @param {object} options - [Configuration object](#/docs/?method=gcloud).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var dns = gcloud.dns();
+ */
+function DNS(options) {
+  if (!(this instanceof DNS)) {
+    return new DNS(options);
+  }
+
+  options = options || {};
+
+  if (!options.projectId) {
+    throw util.missingProjectIdError;
+  }
+
+  this.makeAuthorizedRequest_ = util.makeAuthorizedRequestFactory({
+    credentials: options.credentials,
+    keyFile: options.keyFilename,
+    scopes: SCOPES,
+    email: options.email
+  });
+
+  this.projectId_ = options.projectId;
+}
+
+/**
+ * Create a managed zone.
+ *
+ * @throws {error} If a zone name is not provided.
+ * @throws {error} If a zone dnsName is not provided.
+ *
+ * @param {string} name - Unique name for the zone. E.g. "my-zone"
+ * @param {object} config - Configuration object.
+ * @param {string} config.dnsName - DNS name for the zone. E.g. "example.com."
+ * @param {string=} config.description - Description text for the zone.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {?module:dns/zone} callback.zone - A new {module:dns/zone} object.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * dns.createZone('my-awesome-zone', {
+ *   dnsName: 'example.com.', // note the period at the end of the domain.
+ *   description: 'This zone is awesome!'
+ * }, function(err, zone, apiResponse) {
+ *   if (!err) {
+ *     // The zone was created successfully.
+ *   }
+ * });
+ */
+DNS.prototype.createZone = function(name, config, callback) {
+  var self = this;
+
+  if (!name) {
+    throw new Error('A zone name is required.');
+  }
+
+  if (!config || !config.dnsName) {
+    throw new Error('A zone dnsName is required.');
+  }
+
+  config.name = name;
+
+  // Required by the API.
+  config.description = config.description || '';
+
+  this.makeReq_('POST', '/managedZones', null, config, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var zone = self.zone(resp.name);
+    zone.metadata = resp;
+
+    callback(null, zone, resp);
+  });
+};
+
+/**
+ * Gets a list of managed zones for the project.
+ *
+ * @param {object=} query - Query object.
+ * @param {number} query.maxResults - Maximum number of results to return.
+ * @param {string} query.pageToken - Page token.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {?module:dns/zone[]} callback.zones - An array of {module:dns/zone}
+ *     objects.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * dns.getZones(function(err, zones, apiResponse) {});
+ */
+DNS.prototype.getZones = function(query, callback) {
```

Comment:
How come we call this `getZones` vs. something like `listZones`? The only reason I ask is because I wanted to see what options I had available in query and my initial reaction was to look at ManagedZones > get.